src/lib.rs

```diff
@@ -343,16 +356,17 @@ impl GvasFile {
     /// ```no_run
     /// use gvas::{error::Error, GvasFile};
     /// use std::fs::File;
+    /// use gvas::game_version::GameVersion;
     ///
     /// let mut file = File::open("save.sav")?;
-    /// let gvas_file = GvasFile::read(&mut file);
+    /// let gvas_file = GvasFile::read(&mut file, GameVersion::Default);
     ///
     /// println!("{:#?}", gvas_file);
     /// # Ok::<(), Error>(())
     /// ```
-    pub fn read<R: Read + Seek>(cursor: &mut R) -> Result<Self, Error> {
+    pub fn read<R: Read + Seek>(cursor: &mut R, game_version: GameVersion) -> Result<Self, Error> {
```

Comment:
This will require consumers to change code to pass in a default value. I would suggest making a different function with this functionality, and leaving the original API as it was.

Reply:
This would also require a function like `read_with_game_version_with_hints`, imo the breaking change is justified

Reply:
How about the `read_with_hints` variant? I would be in favor of consolidating all of the `read*` variants in to a single function, which might look something like this:

```rs
    pub fn read<R: Read + Seek>(
        cursor: &mut R,
        game_version: GameVersion,
        hints: Option<&HashMap<String, String>>,
    ) -> Result<Self, Error>
```

I think it should also be possible to improve `read` to auto-detect an appropriate GameVersion value without taking it as an argument.